src/Serokell/Util/Version.hs

```diff
@@ -0,0 +1,43 @@
+module Serokell.Util.Version
+       ( -- * Git revision
+         retrieveGitRev
+       ) where
+
+import Universum
+
+import Instances.TH.Lift ()
+import System.Environment (lookupEnv)
+import System.Exit (ExitCode (..))
+import System.Process (readProcessWithExitCode)
+
+import qualified Data.Text as Text
+import qualified Language.Haskell.TH as TH
+import qualified Language.Haskell.TH.Syntax as TH
+
+{- | Gets the git revision.
+For example, you can use it configuring CLI options:
+
+@
+versionOption = infoOption
+    ("Git revision: " <> toString $(retrieveGitRev))
+    (long "version" <> help "Show version")
+@
+
+Note, that you'll need @-XTemplateHaskell@.
+-}
+retrieveGitRev :: TH.Q TH.Exp
+retrieveGitRev = do
+    cti <- TH.runIO $ Text.strip . fromString <$> retrieveGit
+    TH.lift cti
+  where
+    retrieveGit :: IO String
+    retrieveGit =
+        lookupEnv "GITREV" >>= maybe retrieveFromGitExecutable pure
```

Comment:
You can use [`whenNothingM`](https://hackage.haskell.org/package/universum-1.1.0/docs/Universum-Monad-Maybe.html#v:whenNothingM) from `universum` here. And write this in single line.